hooks/useTokenList.tsx

```diff
@@ -27,8 +28,18 @@ export const useTokenList = () => {
       const tokenList = json.tokens.filter(
         (token) => token.chainId === chainId
       );
-      return tokenList;
+      const weth = tokenList.find((token) => token.symbol === "WETH")?.address;
+      if (!weth) throw new Error("No WETH found in token list");
```

Comment:
Intead of throwing, we can manually define WETH (per-chain, of course, since it's different per-chain). Need to find what the equivalents are for BSC and Polygon, probably WBNB and WMATIC if I had to guess

Reply:
we could do that, but don't we expect WETH in each token list? and if it's not there, mightn't it be grounds for a big error?

Reply:
Hmm, yea in theory each token list should have the wrapped version of the native token (WETH), and we can verify that upfront before committing to that token list. Partly related to https://github.com/ScopeLift/railgun-shielder/issues/43, because if the list is a static, local JSON file this line can neve occur. But if loading the token lists from a URL, it's possible the list changes and removes WETH for whatever reason. In that case we _can_ throw an error, but since we know the address of an immutable WETH contract that will always work, we can just plug that in instead

Reply:
ok, how's this for ya! e7051b8

Reply:
Like it! Just needs a few fixes.

First, all the WETH addresses are `0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2` but that's not correct. For example, `cast code 0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2 --rpc-url $POLYGON_RPC_URL`(or another RPC) shows there is no code at those addresses. Usually you know you've found the right contract because it (1) holds a ton of the native token, and (2) is written in solidity 0.4.18 (or similar) since that's what the original mainnet WETH contract is written in, and everyone copies it

Second, not all of them are called WETH, since WETH = Wrapped Ether and other chains have different native token names, e.g. on matic it's Wrapped Matic / WMATIC

Here's the info you need for each chain, in the format "Name / Symbol / Decimals / Address"
- Mainnet: Wrapped Ether / WETH / 18 / 0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2
- Goerli: Wrapped Ether / WETH / 18 / 0xB4FBF271143F4FBf7B91A5ded31805e42b2208d6 
- BSC: Wrapped BNB / WBNB / 18 / 0xbb4CdB9CBd36B01bD1cBaEBF2De08d9173bc095c
- Polygon: Wrapped Matic / WMATIC / 18 / 0x0d500B1d8E8eF31E21C99d1Db9A6444d3ADf1270
- Arbitrum: Wrapped Ether / WETH / 18 / 0x82aF49447D8a07e3bd95BD0d56f35241523fBab1

Reply:
thanks matt will be fixing in #51

Reply:
going to handle in #51 , thank you!